README.md

```diff
@@ -69,7 +69,45 @@ steps:
 
 > And did you also see how the timeouts for the steps are derived from the commands?
 
-and
+Since version 5 we also support the [new `depends_on` syntax](https://buildkite.com/changelog/84-introducing-pipeline-step-dependencies):
+
+```yaml
+steps:
+    - command:
+          - yarn
+          - yarn lint
+      key: 628a5ef6-ae4d-4ec6-bff2-9e4fb4410162
+      timeout_in_minutes: 3
+    - command:
+          - yarn
+          - yarn build
+      key: 7924d03c-8604-47d4-bdb8-8b25eb6fd77c
+      depends_on:
+          - step: 628a5ef6-ae4d-4ec6-bff2-9e4fb4410162
+      timeout_in_minutes: 7
+    - command:
+          - yarn
+          - yarn test
+      key: 69f7e18a-f2ad-4bbe-9bb8-f09ada40d949
+      depends_on:
+          - step: 628a5ef6-ae4d-4ec6-bff2-9e4fb4410162
+      timeout_in_minutes: 4
+    - command:
+          - yarn
+          - yarn integration
+      key: 48cff6d2-66df-499b-b7ec-6672c684bc6d
+      depends_on:
+          - step: 7924d03c-8604-47d4-bdb8-8b25eb6fd77c
+      timeout_in_minutes: 12
+```
+
+you can get this format by using a different serializer:
+
+```ts
+console.log(await new YamlDependsOnSerializer().serialize(pipeline));
```

Comment:
I think this would be convonient to make it an option to the existing serializer, like:
```suggestion
console.log(await new YamlSerializer({ dependsOn: true }).serialize(pipeline));
```

Reply:
Below I suggested even more(I hope) convinient way to serialize a pipeline. PTAL.

Reply:
^ I went with your first suggestion